### PR TITLE
feat(iam-identity): adding trusted profiles

### PR DESCRIPTION
**Pull request by pooraniss-ibm**
## PR summary
This pr contains all the supported endpoints of Iam-Identity's trusted profiles.
we recently went public and here is the documentation: https://test.cloud.ibm.com/apidocs/iam-identity-token-api

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

## Other information

Screenshot 2021-08-25 at 01 02 09

### CODE REVIEW

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 1]
```diff
@@ -210,6 +220,7 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
 			// begin-get_api_key
 
 			getAPIKeyOptions := iamIdentityService.NewGetAPIKeyOptions(apikeyID)
+            getAPIKeyOptions.SetIncludeHistory(false)
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:223]: Thanks for making this change, and I apologise for being picky. The indentation is off a bit.

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 2]
```diff
@@ -426,6 +438,292 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
...
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(profile).ToNot(BeNil())
+			Expect(profileEtag).ToNot(BeEmpty())
+		})
+		It(`ListProfiles request example`, func() {
+			fmt.Println("\nListProfiles() result:")
+			// begin-list_profile
+
+			listProfileOptions := iamIdentityService.ListProfileOptions()
+			listProfileOptions.SetAccountID(accountID),
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:489]: Please add the `includeHistory:true` attribute too. It is setup across the SDKs and we'd like to achieve consistency across SDK example code snippets.

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 3]
```diff
@@ -426,6 +438,292 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
...
+
+			// end-update_profile
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(profile).ToNot(BeNil())
+		})
+		It(`CreateClaimRule request example`, func() {
+			fmt.Println("\nCreateClaimRule() result:")
+			// begin-create_claimrule
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:526]:
```suggestion
			// begin-create_claimRule
```

Comment (Andras-Csanyi) [reply iamidentityv1/iam_identity_v1_examples_test.go:526]: Its closing tag too.

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 4]
```diff
@@ -426,6 +438,292 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
...
+			// end-get_claimrule
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(claimRule).ToNot(BeNil())
+			Expect(claimRuleEtag).ToNot(BeEmpty())
+		})
+		It(`ListClaimRules request example`, func() {
+			fmt.Println("\nListClaimRules() result:")
+			// begin-list_claimrule
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:574]:
```suggestion
			// begin-list_claimRules
```

Comment (Andras-Csanyi) [reply iamidentityv1/iam_identity_v1_examples_test.go:574]: Its closing tag too

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 5]
```diff
@@ -426,6 +438,292 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
...
+
+			// end-get_link
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(profile).ToNot(BeNil())
+		})
+		It(`ListLinks request example`, func() {
+			fmt.Println("\nListLinks() result:")
+			// begin-list_link
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:678]:
```suggestion
			// begin-list_links
```

Comment (Andras-Csanyi) [reply iamidentityv1/iam_identity_v1_examples_test.go:678]: Its closing tag too

**Comment by pooraniss-ibm** [comment]
this PR is now updated, all the requested changes are made :)

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 6]
```diff
@@ -426,6 +438,293 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
...
+			// end-get_claimRule
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(claimRule).ToNot(BeNil())
+			Expect(claimRuleEtag).ToNot(BeEmpty())
+		})
+		It(`ListClaimRules request example`, func() {
+			fmt.Println("\nListClaimRules() result:")
+			// begin-list_claimRule
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:575]:
```suggestion
			// begin-list_claimRules
```

File: iamidentityv1/iam_identity_v1_examples_test.go [hunk 7]
```diff
@@ -426,6 +438,293 @@ var _ = Describe(`IamIdentityV1 Examples Tests`, func() {
...
+			listClaimRulesOptions.SetProfileID(profileId)
+
+			claimRulesList, response, err := iamIdentityService.ListClaimRules(listClaimRulesOptions)
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(claimRulesList, "", "  ")
+			fmt.Println(string(b))
+
+			// end-list_claimRule
```
Comment (Andras-Csanyi) [inline iamidentityv1/iam_identity_v1_examples_test.go:587]:
```suggestion
			// end-list_claimRules
```

**Review by Andras-Csanyi — APPROVED** [review]
When the PR build finishes and it is green I'll merge it.

**Pull request merged**